app/controllers/front_ui_controller.rb

```diff
@@ -38,6 +38,14 @@ def save_location
     render nothing: true
   end
 
+  def view_map
```

Comment:
Oh, maybe this belongs in the `maps_controller`, next to the `show` method? Thanks, Cess!